### PR TITLE
Refactor cmd/rigtest into a real test suite in test/rig_test.go and fix found bugs

### PR DESCRIPTION
**Pull request by kke**
The `cmd/rigtest` tool was replaced with an actual test suite in `test/rig_test.go`. The improved tests uncovered a bunch of bugs that were fixed in this one as well.

The details of what was fixed were lost while this got quite complex, which is again a sign that the library has become too complicated for its own good and need to be addressed in #92 or other PRs.

### CODE REVIEW

File: pkg/rigfs/winfile.go [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+	_, err := fmt.Fprintf(f.stdin, "%s\n", cmd)
+	if err != nil {
+		return nil, f.pathErr(OP_OPEN, fmt.Errorf("write command: %w", err))
+	}
+	if cmd == "q" {
+		return &rcpResponse{}, nil
+	}
+	select {
+	case <-f.done:
+		return nil, fmt.Errorf("rigrcp exited")
```
Comment (twz123) [inline pkg/rigfs/winfile.go:238]: errors.New

File: ssh.go [hunk 2]
```diff
@@ -617,20 +619,22 @@ func (c *SSH) Exec(cmd string, opts ...exec.Option) error { //nolint:cyclop
...
 	go func() {
 		defer wg.Done()
 		outputScanner := bufio.NewScanner(stderr)
 
 		for outputScanner.Scan() {
-			gotErrors = true
-			execOpts.AddOutput(c.String(), "", outputScanner.Text()+"\n")
+			msg := outputScanner.Text()
+			if msg != "" {
+				errors = append(errors, err.Error())
```
Comment (twz123) [inline ssh.go:632]: Is that the right error to add here? I'm somehow failing to see where this is coming from.

File: os/linux.go [hunk 3]
```diff
@@ -256,11 +257,15 @@ func (c Linux) InstallFile(h Host, src, dst, permissions string) error {
 
 // ReadFile reads a files contents from the host.
 func (c Linux) ReadFile(h Host, path string) (string, error) {
-	out, err := h.ExecOutputf("cat -- %s 2> /dev/null", shellescape.Quote(path), exec.HideOutput(), exec.Sudo(h))
+	out, err := h.ExecOutputf("base64 %s 2> /dev/null", shellescape.Quote(path), exec.HideOutput(), exec.Sudo(h))
```
[the remark this answers is not in the document]
Comment (kke) [reply os/linux.go:260]: this was just a quick temporary fix to avoid it eating trailing linefeeds

File: ssh.go (hunk 2 above)
Comment (kke) [reply ssh.go:632]: no

File: os/linux.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (kke) [reply os/linux.go:260]: (which could be fixed by removing the `strings.TrimSpace` in the ExecOutput function, but that breaks a whole lot of things elsewhere)

[the remark this answers is not in the document]
Comment (kke) [reply os/linux.go:260]: (or by using the ExecStreams variant here)

**Pull request merged**